### PR TITLE
Added method where(spec)

### LINKED ISSUE #2: New Method: where(spec)

### PR DESCRIPTION
**Pull request by islas27**
This pull request closes #2
- Added new method `where(spec)` to `Array.prototype`
- Added the unit test
- Added the documentation
- The example files can run in the terminal with `node example/each.js` for example and receive the same output as described in the comments

### CODE REVIEW

**Comment by islas27** [comment]
Thanks @fhernandezn for the tip!

File: aee.js [hunk 1]
```diff
@@ -5,3 +5,12 @@ Array.prototype.each = function (callback) {
   }
   return this
 }
+
+Array.prototype.where = function (spec) {
+  const len = this.length
+  var newArray = []
+  for (var i = 0; i < len; i += 1) {
+    if (spec(this[i])) newArray.push(this[i])
```
Comment (fhernandezn) [inline aee.js]: According with the specification, you need to pass `i` to the `spec` function

File: aee.js [hunk 2]
```diff
@@ -5,3 +5,12 @@ Array.prototype.each = function (callback) {
   }
   return this
 }
+
+Array.prototype.where = function (spec) {
+  const len = this.length
+  var newArray = []
```
Comment (ryanez) [inline aee.js]: lets call it `var result = []` (or `const`)

File: aee.js (hunk 1 above)
Comment (islas27) [reply aee.js]: fixed in commit 9d85513

**Review by ryanez — COMMENTED** [review]
Which are the the code conventions or style guides that you are using/following?

Speaking for myself I like the airbnb style guides.
https://github.com/airbnb/javascript

**Comment by islas27** [comment]
I use the [Standard](http://standardjs.com/rules.html) style, it seems more readable for me.

File: aee.js (hunk 2 above)
Comment (islas27) [reply aee.js]: alright, will do.
Edit: fixed in commit 63a9262

**Comment by ryanez** [comment]
It's OK, if you want to use it for your own projects (like this), but be prepared to follow the conventions your current team decides to use.

**Comment by ryanez** [comment]
LGMT 👍

**Pull request merged**